### PR TITLE
Multiple filter w/ include fix + internal routing fix + more

### PR DESCRIPTION
**Pull request by st3xupery**
Applied a fix that was breaking the ability to acquire includes when using multiple filters through a comma separated list. It seems I was splitting the filter param by comma too far into a particular function that wasn't called when internally routing includes. You'll see the filter split happens immediately in the parseandValidate function. I wrote up two tests for this and added an additional photo resource which required me to modify the counts in a few test assertions.

Additionally in the include lib when building a list of resources to fetch I reduced the url to just the base + path of the url as the hostname is not required while routing internally. This also remedies a problem where the use of pathPrefix which is overwritten by the apiPrefixOverride would cause this route to be incorrect from an internal standpoint.

Finally I appended originalUrl to the request object which persists as the same value during rerouting as it allows me to compare the route object to the originalUrl to see if the request is the result of an include or whatever else may prompt a reroute. I feel like this is a quasi acceptable use of the param as defined in the express API here http://expressjs.com/en/api.html#req.originalUrl

### CODE REVIEW

**Comment by st3xupery** [comment]
There was also an issue where setting graphiql as false would still run the joiconverter process to build the definition. So now if the param is false it skips executing the process.

The API I'm building has attributes that are objects which throws errors with the Joi to GraphQL converter so maybe it's something to be fixed, but in the interim I'd just like to have it off.

**Comment by theninj4** [comment]
@pmcnr-hx would you like to take a review on this one? You were the closest to the filter preprocessing a few months back.

I'm fine with the idea of turning off all of GraphQL with a strict `false` in the config :+1:

**Comment by pmcnr-hx** [comment]
I'll be reviewing this soon, but the changes are not trivial, so I'll need to take some time.

**Comment by st3xupery** [comment]
Is there anything I can do to help you along?

**Comment by pmcnr-hx** [comment]
I'll review today @st3xupery. Meanwhile can you merge or rebase on master?

**Comment by st3xupery** [comment]
@pmcnr-hx By that what do you mean. Do you want me to make my pull request off of the master of my fork? I'm still using the master for legacy support of one of my projects. I can make another fork if need be.

**Comment by pmcnr-hx** [comment]
@st3xupery. What you just did. Thanks. :) I'm moving this to the top of my TODO list. Should be able to get round to it soon. Apologies for the delay.

**Comment by st3xupery** [comment]
@pmcnr-hx Okay, great and no worries~!

File: lib/filter.js [hunk 1]
```diff
@@ -117,6 +115,8 @@ filter.parseAndValidate = request => {
   for (const key in request.params.filter) {
     filterElement = request.params.filter[key]
 
+    if (typeof filterElement === 'string') request.params.filter[key] = filterElement = filterElement.split(FILTER_SEPERATOR)
```
Comment (pmcnr-hx) [inline lib/filter.js]: This is a pragmatic solution, but I'm not too thrilled about mutating `request.params.filter[key]`. I'm going to have a look at the internal routing and check if we can used the processed filter there also.

File: lib/graphQl/index.js [hunk 2]
```diff
@@ -13,10 +13,12 @@ const writeTypes = require('./writeTypes.js')
 jsonApiGraphQL.with = app => {
   const config = jsonApi._apiConfig
 
-  app.use(new RegExp(`${config.base}$`), graphqlHTTP({
-    schema: jsonApiGraphQL.generate(jsonApi._resources),
-    graphiql: !!config.graphiql
-  }))
+  if (config.graphiql !== false) {
```
Comment (pmcnr-hx) [inline lib/graphQl/index.js]: Shall we introduce the `strict` configuration property that @theninj4 [suggests](https://github.com/holidayextras/jsonapi-server/pull/233#issuecomment-269498049) to turn off the GraphQL endpoint? Abusing the `graphiql` configuration property for this end is not the best option.

File: lib/swagger/index.js [hunk 3]
```diff
@@ -17,7 +17,8 @@ swagger._getSwaggerBase = () => {
   let basePath, host, protocol
   if (jsonApi._apiConfig.urlPrefixAlias) {
     const urlObj = url.parse(jsonApi._apiConfig.urlPrefixAlias)
-    basePath = urlObj.pathname.replace(/\/$/, '')
```
Comment (pmcnr-hx) [inline lib/swagger/index.js]: If we use the `(?<!^)\/$` regexp (with a look behind to assert we're not at the beginning of the string) above we can avoid the `if`.

**Comment by yossialush** [comment]
Hi, this fix is kind of crucial for us. It's blocking us from upgrading to latest version, thus we stuck with old bugs. When should we anticipate for this merge to take place?

**Comment by theninj4** [comment]
It seems like there are no big objections to what's going on here. As soon as the build goes green, I'll merge and release. Looks like the last commit broke something, it looks unrelated to the rest of this pr, so if the build is still failing tomorrow midday GMT I might merge this PR then revert the troublesome commit.

**Comment by yossialush** [comment]
Good news! Thanks 🎉

**Comment by st3xupery** [comment]
Hey! Sorry I kind of neglected this. 
The last commit was a fix to a whole other issue. I'll revert this branch to the previous commit and make a pull on the other issue separately. 
As far as @pmcnr-hx are concerned. I'm not too clear on the 2nd, isn't technically !== false a strict equivalency, and I can implement the regexp expression in his 3rd suggestion.
I'll do this all by tonight EST.

**Comment by st3xupery** [comment]
I've made the regex replacement of the if statement. Had to use something slightly different as javascript does not have a *look behind* in it's regex capabilities. The alternative works to the same effect: `(?!^\/)\/$`

File: lib/swagger/index.js (hunk 3 above)
Comment (st3xupery) [inline lib/swagger/index.js]: Had to use something slightly different as javascript does not have a look behind in it's regex capabilities. The alternative works to the same effect: (?!^\/)\/$

File: lib/graphQl/index.js (hunk 2 above)
Comment (st3xupery) [inline lib/graphQl/index.js]: Not sure what you mean, is !== false not a strict equivalency?

Comment (pmcnr-hx) [inline lib/graphQl/index.js]: What we were suggesting was a *new* configuration property named `strict`, but we can get round to doing that in a later PR.

**Pull request merged**